src/components/Channel/Channel.tsx

```diff
@@ -609,12 +603,12 @@ export const ChannelWithContext = <
   const copyChannelState = lightThrottle(() => {
     setLoading(false);
     if (channel) {
-      setMembers(channel.state.members);
-      setMessages(channel.state.messages);
-      setRead(channel.state.read);
-      setTyping(channel.state.typing);
+      setMembers({ ...channel.state.members });
+      setMessages([...channel.state.messages]);
+      setRead({ ...channel.state.read });
+      setTyping({ ...channel.state.typing });
       setWatcherCount(channel.state.watcher_count);
-      setWatchers(channel.state.watchers);
+      setWatchers({ ...channel.state.watchers });
```

Comment:
@virdesai Can we plz do this on js client level? I mean not mutating the object in js client